### DISCUSSION TITLE
Could you possibly provide a list of the most often banned domains?

### DISCUSSION THREAD

**Discussion by ghost** [EFForg/privacybadger#2786] [Show and tell]
Hello,
Could you kindly supply us with a report or a list of the most often banned domains?
- prajwal

**Comment by ghostwords** [comment]
Could you explain what you are looking to accomplish? My first thought is that we don't have this information (since Privacy Badger does not automatically collect any data from users).

**Reply by ghost** [reply]
We can block these domains on the DNS level instead of banning them on the browser level, also collect data on which domains were blocked.

For anyone who want to avoid being tracked, this information would be priceless.

**Comment by ghostwords** [comment]
Take a look at [Badger Sett](https://github.com/EFForg/badger-sett), the Web crawler we use to train Privacy Badger. See [seed.json](https://github.com/EFForg/privacybadger/blob/master/src/data/seed.json) in this repository for the latest pre-trained list used by Privacy Badger. See [here for an explanation of the data format](https://github.com/EFForg/privacybadger/blob/master/doc/DESIGN-AND-ROADMAP.md#data-structures).

The thing though is that Privacy Badger does much more than just block domains from a list. To start with:
- Some domains should not be blocked outright as blocking them causes too much breakage. Privacy Badger [denies them access to cookies and local storage](https://privacybadger.org/#Does-Privacy-Badger-contain-a-list-of-blocked-sites) as a compromise.
- Some other domains also break many websites when fully blocked, but Privacy Badger knows to replace scripts served by those domains with [local versions of those scripts with all tracking removed](https://github.com/EFForg/privacybadger/blob/master/src/data/surrogates.js). This resolves breakages and maintains full privacy.
- Many domains are in use on websites that belong to the same corporate entity (first-party) as well as on unaffiliated (third-party) websites. Privacy Badger keeps [lists of affiliated domains](https://github.com/EFForg/privacybadger/blob/master/src/js/multiDomainFirstParties.js) and uses them to only learn to block domains on those lists in third-party contexts.
- Some domains provide potentially useful widgets (video players, comments, etc.). They shouldn't be allowed to track you by default, but [you should be able to allow them selectively](https://privacybadger.org/#How-does-Privacy-Badger-handle-social-media-widgets).
- Privacy Badger can reduce the tracking on some sites you visit directly. Sites like [Facebook](https://www.eff.org/deeplinks/2018/05/privacy-badger-rolls-out-new-ways-fight-facebook-tracking) and [Google](https://www.eff.org/deeplinks/2018/10/privacy-badger-now-fights-more-sneaky-google-tracking) track the links you click that take you elsewhere. Privacy Badger tries to block this tracking.
- If you browse a portion of the Web poorly served by existing ad blocker lists, you can enable Privacy Badger's learning and benefit from your Privacy Badger automatically learning to block as-yet-unknown domains that appear to track you across the sites you visit.
- Privacy Badger sends websites the [Global Privacy Control](https://globalprivacycontrol.org/) and Do Not Track signals. By doing this, you join an effort to make the Web better for everybody.

Let me know if you have any questions.
